summary/usergroup.go

```diff
@@ -47,20 +48,10 @@ type dirStore map[string]*summary
 
 // addForEachDir breaks path into each directory and calls add() on it.
 func (store dirStore) addForEachDir(path string, size int64) {
-	cb := func(dir string) {
-		store.add(dir, size)
-	}
-
-	doForEachDir(path, cb)
```

Comment:
How does this work without calling it for every parent dir?

Reply:
We embedded `doForEachDir` in the func to avoid allocating the callback func.